qemu/qmp.go

```diff
@@ -883,11 +891,31 @@ func (q *QMP) ExecHotplugMemory(ctx context.Context, qomtype, id, mempath string
 }
 
 // ExecutePCIVSockAdd adds a vhost-vsock-pci bus
-func (q *QMP) ExecutePCIVSockAdd(ctx context.Context, id, guestCID string) error {
+func (q *QMP) ExecutePCIVSockAdd(ctx context.Context, id, guestCID, vhostfd string, disableModern bool) error {
 	args := map[string]interface{}{
 		"driver":    VHostVSockPCI,
 		"id":        id,
 		"guest-cid": guestCID,
 	}
+
+	if vhostfd != "" {
+		args["vhostfd"] = vhostfd
+	}
+
+	if disableModern {
+		args["disable-modern"] = disableModern
+	}
+
 	return q.executeCommand(ctx, "device_add", args, nil)
 }
+
+// ExecuteGetFD sends a file descriptor via SCM rights and assigns it a name
+func (q *QMP) ExecuteGetFD(ctx context.Context, fdname string, fd uintptr) error {
```

Comment:
Same here.

Reply:
nop, `fdname` can be an ID used to identify the file descriptor in the VM

Reply:
What about `fd uintptr`?